File: services/customers-service/src/main/java/br/com/deltaglobalbank/customers/domain/document/DocumentType.java
package br.com.deltaglobalbank.customers.domain.document;

import java.util.Arrays;
import java.util.Locale;

public enum DocumentType {
    RG,
    CNH;

    public String toDatabaseValue() {
        return name().toLowerCase(Locale.ROOT);
    }

    public static DocumentType fromDatabaseValue(String value) {
        return Arrays.stream(values())
            .filter(type -> type.name().equalsIgnoreCase(value))
            .findFirst()
            .orElseThrow(() -> new IllegalArgumentException("invalid DocumentType: " + value));
    }
}

File: services/customers-service/src/main/kotlin/br/com/deltaglobalbank/customers/domain/document/DocumentType.kt
package br.com.deltaglobalbank.customers.domain.document

enum class DocumentType {
    RG,
    CNH;

    fun toDatabaseValue() : String = name.lowercase()

    companion object {
        fun fromDatabaseValue(value: String): DocumentType =
            DocumentType.entries.firstOrNull{ it.name.equals(value, ignoreCase = true) }
                ?: throw IllegalArgumentException("invalid DocumentType: $value")
    }
}